include/region_map.h

```diff
@@ -794,14 +793,14 @@ namespace region_map_detail {
 
 		bool is_underfull() const { return m_children.size() < min_children; }
 
-		GridBox<Dims> sanity_check_bounding_boxes() const {
+		box<Dims> sanity_check_bounding_boxes() const {
 #if !defined(NDEBUG)
-			// After an erase this node might not have any children. Return empty box in that case.
-			if(m_child_boxes.empty()) { return box_cast<Dims>(GridBox<3>({0, 0, 0}, {0, 0, 0})); }
+			// After an erase this node might not have any children. Return empty box in that case. TODO this breaks for Dims == 0 (where area is always 1)!
```

Comment:
I would like to see the appropriate `region_map` refactoring in a separate PR so we can compare correctness / performance (see my "future work" in the OP).

Reply:
I was a bit confused at first because we create an N-dimensional region map for N-dimensional buffers, but I forgot that we still have [this specialization](https://github.com/celerity/celerity-runtime/blob/e6a16abb12d8586691db6007db01447b4c72b618/include/region_map.h#L1226) for Dims == 0.

> I would like to move the region_map refactoring to a separate PR so we can compare correctness / performance (see my "future work" in the OP).

Okay, sure.

Reply:
As discussed in person, we should probably keep the impl specialization for 0d region maps, but all its internal interfaces should be adapted operate on 0d boxes.